dali/python/backend_impl.cc

```diff
@@ -1569,13 +1577,18 @@ PYBIND11_MODULE(backend_impl, m) {
                                       (&Pipeline::AddOperator))
     .def("GetOperatorNode", &Pipeline::GetOperatorNode)
     .def("Build",
-        [](Pipeline *p, const std::vector<std::pair<string, string>>& outputs) {
-          p->Build(outputs);
-          })
+         [](Pipeline *p, const std::vector<std::pair<string, string>> &outputs) {
+             p->Build(outputs);
+         })
     .def("Build",
-        [](Pipeline *p) {
-          p->Build();
-          })
+         [](Pipeline *p, const std::vector<OutputDesc> &outputs) {
+             std::vector<PipelineOutputDesc> build_args;
+             for (auto& out : outputs) {
+               build_args.emplace_back(to_struct<PipelineOutputDesc>(out));
+             }
+             p->Build(build_args);
+         })
```

Comment:
I know you've kept the overload in C++ for the purpose of not reworking the C++ API tests, but maybe we should keep only one option in Python. This is fully internal and used in one place, maybe remove the old overload?